web-search-engines/langchain4j-web-search-engine-searchapi/src/main/java/dev/langchain4j/web/search/searchapi/SearchApiWebSearchEngine.java

```diff
@@ -0,0 +1,277 @@
+		
+		if (DEFAULT_ENGINE.equalsIgnoreCase(this.engine)) {
+			final Map<String, Object> params = addDefaultSearchParameters(request);
+			final Locale locale = new Locale(webSearchRequest.language());
+	        params.put("hl", locale.hl);
+	        params.put("gl", locale.gl);
+		}
+	}
+
+    /**
+     * This life cycle method is always called after a search is performed by <code>WebSearchEngine#search</code> 
+     * but before the results are returned to the caller.
+     * 
+     * Future search engines must call this method to customize the search results. 
+	 * Subclasses should use this method to apply transformations to the <code>List<WebSearchOrganicResult> results<code> 
+	 * prior to it being sent to the caller.
+     * 
+     * @param response
+     * @param results
+     */
+	@SuppressWarnings("unchecked")
+	protected void customizeSearchResults(final SearchApiResponse response, final List<WebSearchOrganicResult> results) {
+		
+		if (!DEFAULT_ENGINE.equalsIgnoreCase(this.engine)) {
+			return;
+		}
+		// For the Google Search engine, searchapi.io may include 1 or more optional JSON elements in the response
+		// that directly answers the search query in the "knowledge_graph",
+		// "answer_box", etc
+		// such optional elements counts toward the "num" parameter. For instance, if
+		// "num" is set to 5 and an "inline_images" is included, then "organic_results"
+		// will contain 4 (instead of 5) items,
+		// which is why we have to add it back to "results" below as the first item.
+		// See also:
+		// https://api.python.langchain.com/en/latest/_modules/langchain_community/utilities/searchapi.html#SearchApiAPIWrapper
+
+		String title = null;
+		String website = DEFAULT_BASE_URL;
+		String description = null;
+
+		if (response.getAnswerBox() != null && !response.getAnswerBox().isEmpty()) {
+			title = (String) response.getAnswerBox().get("title");
+			website = (String) response.getAnswerBox().get("link");
+
+			if (response.getAnswerBox().containsKey("answer")) {
+				description = (String) response.getAnswerBox().get("answer");
+
+			} else if (response.getAnswerBox().containsKey("snippet")) {
+				description = (String) response.getAnswerBox().get("snippet");
+			}
+		} else if (response.getKnowledgeGraph() != null && !response.getKnowledgeGraph().isEmpty()) {
+			title = (String) response.getKnowledgeGraph().get("title");
+			website = response.getKnowledgeGraph().containsKey("website")
+					? (String) response.getKnowledgeGraph().get("website")
+					: website;
+			description = (String) response.getKnowledgeGraph().get("description");
+
+		} else if (response.getInlineVideos() != null && !response.getInlineVideos().isEmpty()) {
+			final Map<String, Object> video = response.getInlineVideos().get(0);
+			title = (String) video.get("title");
+			website = video.containsKey("link") ? (String) video.get("link") : website;
+			description = (String) video.get("image");
```

Comment:
I'm not sure if `inlineVideos` should be included in this first version. We must include web page results and images results only.